src/basic/siphash24.h

```diff
@@ -17,9 +17,15 @@ struct siphash {
 
 void siphash24_init(struct siphash *state, const uint8_t k[static 16]);
 void siphash24_compress(const void *in, size_t inlen, struct siphash *state);
-void siphash24_compress_boolean(bool in, struct siphash *state);
 #define siphash24_compress_byte(byte, state) siphash24_compress((const uint8_t[]) { (byte) }, 1, (state))
 
+static inline void siphash24_compress_boolean(bool in, struct siphash *state) {
+        int i = in;
+
+        siphash24_compress(&i, sizeof i, state);
```

Comment:
hmm, this hashes to a different value and becomes endianness dependant, as you hash a native endian 32bit integer here... Why not stick to uin8_t here so that we hash to the same value everywhere?